### ISSUE TITLE
[BUG] Some parts of a trace don't disappear when a new figure is set in dcc.Graph.

### ISSUE THREAD

**Issue by x-tabdeveloping** [plotly/dash#2469]
I'm using Ubuntu 20.04, Chrome 111.0.5563.64 and the following:

```
dash==2.7.1
dash-core-components==2.0.0
dash-daq==0.5.0
dash-extensions==0.1.10
dash-html-components==2.0.0
dash-mantine-components==0.11.1
dash-table==5.0.0
```

## Description
I'm working on a tool right now that displays network graphs in Dash.
I have noticed, that whenever I tried to update the dcc.Graph some of the lines from the previous figure remained on screen,
and they also follow your viewport unlike the rest of the figure. 

## Minimal reproducible example

First I import all things necessary and create an app with a button and a graph on screen:
```python
from typing import Iterable, Optional

import numpy as np
import plotly.graph_objects as go
from dash import Dash, Input, Output, dcc, html

app = Dash()

app.layout = html.Div(
    [
        html.Button(id="button", children="Click me!", n_clicks=0),
        dcc.Graph(
            id="figure",
            style={"flex": 1},
        ),
    ],
    style={
        "display": "flex",
        "position": "fixed",
        "width": "100%",
        "height": "100%",
    },
)
```
I have a function that produces positions with lines in accordance with the [network graph guide](https://plotly.com/python/network-graphs/). For simplicity's sake I just connect all nodes in the graph.
It takes either the x or y positions of the nodes and then returns the x or y positions of the edges.
```python
def edge_pos(pos: Iterable[float]) -> list[Optional[float]]:
    res = []
    for start_pos in pos:
        for end_pos in pos:
            res.extend([start_pos, end_pos, None])
    return res
```
Then I declare a callback that draws a new random network whenever the button is clicked.
```python
@app.callback(Output("figure", "figure"), Input("button", "n_clicks"))
def update_fig(_: int) -> go.Figure:
    # Randomly generate points from a standard normal
    x, y = np.random.normal(0, 1, size=(2, 5))
    # Create the trace for nodes
    node_trace = go.Scatter(x=x, y=y, mode="markers")
    # Create a trace of lines connecting them
    edge_trace = go.Scatter(x=edge_pos(x), y=edge_pos(y), mode="lines")
    fig = go.Figure(
        data=[edge_trace, node_trace],
    )
    return fig
```
And launch the app when the script is run:
```python
if __name__ == "__main__":
    app.run_server(debug=True)
```
If I run the app, and then click the button to generate new plots, some of the edges from previous figures stay on screen but they don't get smaller when zooming out and they follow you when you pan.
If I turn the trace off and on again they disappear, but then if I pan again they reappear.
I checked and this behavior does not occur when I just `.show()` on the figure produced by the callback, so it's exclusive to dcc.Graph.

https://user-images.githubusercontent.com/13087737/226337914-21b4e58d-8d77-4766-903c-0f21fa9b3281.mp4

## Expected Behavior
Everything that was previously on the graph should be cleared before the new graph is displayed.

**Comment by nickmelnikov82** [comment]
One possible solution is to add the "figure" parameter to the "dcc.Graph" property. This parameter will clear the previous data before displaying the new graph. Change your code as follows:
```
dcc.Graph(
id="figure",
figure={}, # Add a "figure" parameter with an empty dictionary
style={"flex": 1},
)
```
This ensures that the previous figure is cleared before the new one is shown.
Also try adding the animate=True parameter to dcc.Graph.
But on my local machine (Ubuntu | Chrome), your code works fine even without the animate and figure parameters, so another possible reason could be outdated libraries or an unupdated browser cache

**Comment by alexcjohnson** [comment]
I suspect this is the bug that was fixed in plotly.js [v2.17.1](https://github.com/plotly/plotly.js/releases/tag/v2.17.1) and included in Dash [v2.8.0](https://github.com/plotly/dash/releases/tag/v2.8.0) and later. @x-tabdeveloping are you still seeing this if you update to the latest version of Dash?

**Comment by gvwilson** [comment]
Hi - we are tidying up stale issues and PRs in Plotly's public repositories so that we can focus on things that are most important to our community. If this issue is still a concern, please add a comment letting us know what recent version of our software you've checked it with so that I can reopen it and add it to our backlog. (Please note that we will give priority to reports that include a short reproducible example.) If you'd like to submit a PR, we'd be happy to prioritize a review, and if it's a request for tech support, please post in our community forum. Thank you - @gvwilson
